### PR TITLE
make sure that wine prefix name is valid

### LINKED ISSUE #226: some characters seem to cause issues in prefix names
When I wrote the script for "Q.U.B.E", I had the impression that Steam doesn't like "." in the folder name. It crashed all the time with `double free or corruption`. I also noticed this error every now and then for other games. Maybe ® etc. are also bad.

We should write a function which converts any given name into a proper prefix folder name.

### PR DESCRIPTION
**Pull request by plata**
Remove all characters which are not:
* 0-9
* a-z
* minus
* space

Fixes #226

### CODE REVIEW

**Comment by madoar** [comment]
I'm not sure if it's such a good idea to simply remove them. I think it would be better to replace them with some other character. I guess it's possible to find an application, whose name is made up only of "illegal" characters.

**Comment by madoar** [comment]
In addition what about `/` and `\`?

**Comment by plata** [comment]
Could happen if the application name is Japanese/Russian... No idea how to handle that. `/` and `\` should be removed.

**Comment by madoar** [comment]
Could we use the `URLEncoder`?

**Comment by plata** [comment]
Wouldn't that produce a lot of `%20` etc.?

**Comment by madoar** [comment]
Is that bad?

**Comment by plata** [comment]
I don't know. Not quite sure which characters can cause such problems.

**Comment by plata** [comment]
Also the prefix names would not really be human readable anymore.

**Comment by madoar** [comment]
I'm not sure this is a real problem if the prefix names aren't directly readable anymore. Often you get a hint, because some fragments should still stay the same.

**Comment by plata** [comment]
At least the prefix name is shown in the GUI. E.g. when you uninstall. I don't think we should show "Assassin_s%20Creed_Unity" there.

**Comment by madoar** [comment]
For that you can use `URLDecoder` :)

**Comment by madoar** [comment]
But I agree, I'm not sure if we really want to do that, because:

1. It's not really nice, and easily forgettable, to add `URLDecoder` everywhere
2. It may confuse the user if he's looking for the prefix folder and it's named differently to what he's shown in POL 5.

**Comment by plata** [comment]
Yep. Would like to hear @qparis opinion about this.

**Comment by plata** [comment]
@madoar @qparis unless you have any better idea I would like to merge this.

File: Functions/Engines/Wine/script.js [hunk 1]
```diff
@@ -111,7 +111,7 @@ Wine.prototype.prefix = function (prefix) {
     }
 
     // set
-    this._prefix = prefix;
+    this._prefix = prefix.replace(/[^a-z0-9_\-\ ]/gi, '');
```
Comment (madoar) [inline Functions/Engines/Wine/script.js]: Just asking: aren't there `"` missing around the regex expression?

Comment (plata) [inline Functions/Engines/Wine/script.js]: No. https://www.w3schools.com/jsref/jsref_replace.asp

Comment (madoar) [inline Functions/Engines/Wine/script.js]: Oh ok, I thought the Java `replace` method was used here, therefore I was a little confused, but it seems to be a specific javascript method instead, interesting.

**Comment by madoar** [comment]
I think we can merge this

**Pull request merged**